org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoWindow.java

```diff
@@ -2225,7 +2225,24 @@ protected int getADWindowID() {
 		if (!isLookup() && Util.isEmpty(isSOTrx)) {
 			isSOTrx = "Y";
 		}
-		
+
+		if (infoWindow != null) {
+
+			MUserDefInfo userDef = MUserDefInfo.getBestMatch(Env.getCtx(), infoWindow.getAD_InfoWindow_ID());
+
+			if (userDef != null) {
+				if (isSOTrx.equals("Y") && userDef.getAD_Window_ID() > 0)
+					return userDef.getAD_Window_ID();
+				else if (isSOTrx.equals("N") && userDef.getPO_Window_ID() > 0)
+					return userDef.getPO_Window_ID();	
+			}
+
+			if (isSOTrx.equals("Y") && infoWindow.getAD_Window_ID() > 0)
+				return infoWindow.getAD_Window_ID();
+			else if (isSOTrx.equals("N") && infoWindow.getPO_Window_ID() > 0)
+				return infoWindow.getPO_Window_ID();				
```

Comment:
for isSOTrx.equals("N"), we should return infoWindow.getAD_Window_ID() if infoWindow.getPO_Window_ID()== 0 and infoWindow.getAD_Window_ID() > 0

Reply:
done